sql/api/src/main/scala/org/apache/spark/sql/streaming/StatefulProcessorHandle.scala

```diff
@@ -43,4 +43,15 @@ private[sql] trait StatefulProcessorHandle extends Serializable {
 
   /** Function to return queryInfo for currently running task */
   def getQueryInfo(): QueryInfo
+
+  /**
+   * Creates new or returns existing list state associated with stateName.
+   * The ListState persists values of type T.
+   * @param stateName - name of the state variable
+   * @param keyEncoder - Spark SQL Encoder for key
+   * @tparam K - type of key
+   * @tparam T - type of state variable
+   * @return - instance of ListState of type T that can be used to store state persistently
+   */
+  def getListState[K, T](stateName: String, keyEncoder: Encoder[K]): ListState[T]
```

Comment:
You can merge back from master once ? we are passing `keyEncoder` through the operator now - so won't need to add it here

Reply:
Done.